src/providers/tfc/planProvider.ts

```diff
@@ -0,0 +1,424 @@
+
+      if (error instanceof ZodiosError) {
+        handleZodiosError(error, message, this.outputChannel, this.reporter);
+        return;
+      }
+
+      if (axios.isAxiosError(error)) {
+        if (error.response?.status === 401) {
+          handleAuthError();
+          return;
+        }
+
+        if (isErrorFromAlias(apiClient.api, 'listRuns', error)) {
+          message += apiErrorsToString(error.response.data.errors);
+          vscode.window.showErrorMessage(message);
+          this.reporter.sendTelemetryException(error);
+          return;
+        }
+      }
+
+      if (error instanceof Error) {
+        message += error.message;
+        vscode.window.showErrorMessage(message);
+        this.reporter.sendTelemetryException(error);
+        return;
+      }
+
+      if (typeof error === 'string') {
+        message += error;
+      }
+      vscode.window.showErrorMessage(message);
+      return;
+    }
+  }
+
+  dispose() {
+    //
+  }
+}
+
+interface PlanLog {
+  plannedChanges?: Change[];
+  changeSummary?: ChangeSummary;
+  driftChanges?: Change[];
+  driftSummary?: DriftSummary;
+  outputs?: Outputs;
+  diagnostics?: Diagnostic[];
+  diagnosticSummary?: DiagnosticSummary;
+}
+
+interface ItemWithChildren {
+  getChildren(): vscode.TreeItem[];
+}
+
+function isItemWithChildren(element: object): element is ItemWithChildren {
+  return 'getChildren' in element;
+}
+
+class PlannedChangesItem extends vscode.TreeItem implements ItemWithChildren {
+  constructor(private plannedChanges: Change[], summary?: ChangeSummary) {
+    let label = 'Planned changes';
+    if (summary) {
+      const labels: string[] = [];
+      if (summary.import > 0) {
+        labels.push(`${summary.import} to import`);
+      }
+      if (summary.add > 0) {
+        labels.push(`${summary.add} to add`);
+      }
+      if (summary.change > 0) {
+        labels.push(`${summary.change} to change`);
+      }
+      if (summary.remove > 0) {
+        labels.push(`${summary.remove} to destroy`);
+      }
+      if (labels.length > 0) {
+        label = `Planned changes: ${labels.join(', ')}`;
+      }
+    }
+    super(label, vscode.TreeItemCollapsibleState.Expanded);
+  }
+
+  getChildren(): vscode.TreeItem[] {
+    const items: vscode.TreeItem[] = [];
+    this.plannedChanges.forEach((change) => {
+      items.push(new PlannedChangeItem(change));
+    });
+    return items;
+  }
+}
+
+class PlannedChangeItem extends vscode.TreeItem {
+  constructor(public change: Change) {
+    let label = change.resource.addr;
+    if (change.previous_resource) {
+      label = `${change.previous_resource.addr} → ${change.resource.addr}`;
+    }
+
+    super(label, vscode.TreeItemCollapsibleState.None);
+    this.id = change.action + '/' + change.resource.addr;
+    const icon = GetChangeActionIcon(change.action);
+    this.iconPath = icon;
+    this.description = change.action;
+
+    const tooltip: vscode.MarkdownString = new vscode.MarkdownString();
```

Comment:
```suggestion
    const tooltip = new vscode.MarkdownString();
```
The type will be inferred from the return value.